shared/src/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -677,8 +677,7 @@ private void notifyDocumentEnded(DocumentReplication update)
     }
 
     private boolean validationFunction(String docID, int flags, long dict, boolean isPush) {
-        Map<String, Object> dictionary = new FLDict(dict).asDict();
-        MutableDocument document = new MutableDocument(docID, dictionary);
+        Document document = new Document(config.getDatabase(), docID, new FLDict(dict));
```

Comment:
is there any impact of changing the type from MutableDocument to Document?

Reply:
MutableDocument is derived from Document and since it now requires to use lazy deserialization for replication filters so initialize from Document is probably the right way..